packages/ipfs-core/src/components/name/index.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+const createPublishAPI = require('./publish')
+const createResolveAPI = require('./resolve')
+const PubSubAPI = require('./pubsub')
+class NameAPI {
+  /**
+   * @param {Object} config
+   * @param {IPNS} config.ipns
+   * @param {PeerId} config.peerId
+   * @param {Options} config.options
+   * @param {DagReader} config.dagReader
+   * @param {IsOnline} config.isOnline
+   * @param {Keychain} config.keychain
+   * @param {DNS} config.dns
+   */
+  constructor ({ dns, ipns, dagReader, peerId, isOnline, keychain, options }) {
+    this.publish = createPublishAPI({ ipns, dagReader, peerId, isOnline, keychain })
+    this.resolve = createResolveAPI({ dns, ipns, peerId, isOnline, options })
+    this.pubsub = new PubSubAPI({ ipns, options: options.EXPERIMENTAL })
+  }
+}
+module.exports = NameAPI
+
+/**
+ * @typedef {ResolveOptions & ExperimentalOptions} Options
+ *
+ * @typedef {Object} ExperimentalOptions
+ * @property {PubSubOptions} [EXPERIMENTAL]
+ *
+ * @typedef {import('./pubsub').Options} PubSubOptions
+ * @typedef {import('./resolve').ResolveOptions} ResolveOptions
```

Comment:
can't see the usage of these

Reply:
Line 26 defines Options as `ResolveOptions & ExperimentalOptions` which is then used in `config.options`.